pylib/Tools/Build/Shell.py

```diff
@@ -132,34 +134,58 @@ def execute(self, log, keyvals, testDef):
                     log['status'] = 1
                     log['stderr'] = "Parent",cmds['parent'],"log not found"
                     return
-            else:
-                log['status'] = 1
-                log['stderr'] = "Parent log not recorded"
-                return
-
         except KeyError:
             log['status'] = 1
             log['stderr'] = "Parent not specified"
             return
         try:
-            location = parentlog['location']
+            parentloc = os.path.join(os.getcwd(),log['options']['scratch'])
+            location = parentloc
         except KeyError:
             log['status'] = 1
-            log['stderr'] = "Location of package to build was not specified in parent stage"
+            log['stderr'] = "No scratch directory in log"
             return
+        if parentlog is not None:
+            try:
+                parentloc = parentlog['location']
+                location = parentloc
+            except KeyError:
+                log['status'] = 1
+                log['stderr'] = "Location of package to build was not specified in parent stage"
+                return
+        else:
+            try:
+                if log['section'].startswith("TestGet:") or log['section'].startswith("MiddlewareGet:"):
+                    location = os.path.join(parentloc,log['section'].replace(":","_"))
+            except KeyError:
+                log['status'] = 1
+                log['stderr'] = "No section in log"
+                return
         # check to see if this is a dryrun
         if testDef.options['dryrun']:
             # just log success and return
             log['status'] = 0
             return
 
+        # Check to see if this needs to be ran if ASIS is specified
         try:
-            if cmds['asis'] and os.path.exists(os.path.join(location,cmds['binary'])) \
-                            and os.path.isfile(os.path.join(location,cmds['binary'])):
-                testDef.logger.verbose_print("As-Is binary " + os.path.join(location,cmds['binary']) + " exists and is a file")
-                log['location'] = location
-                log['status'] = 0
-                return
+            if cmds['asis']:
+                if cmds['asis_target'] is not None:
+                    if os.path.exists(os.path.join(parentloc,cmds['asis_target'])):
+                        testDef.logger.verbose_print("asis_target " + os.path.join(parentloc,cmds['asis_target']) + " exists. Skipping...")
+                        log['location'] = location
+                        log['status'] = 0
```

Comment:
location is updated already to be section name by that point if in the TestGet or MiddlewareGet stage. Otherwise, the location is inherited by the parent